packages/client-utils/src/signing-smart-wallet-kit.ts

```diff
@@ -93,24 +115,83 @@ export const makeSigningSmartWalletKit = async (
     );
 
     const txBytes = TxRaw.encode(signedTx).finish();
-    return client.broadcastTx(txBytes);
+    const tx = await client.broadcastTx(txBytes);
+    if (tx.code === 0) lastTx = tx;
```

Comment:
this is the last _successful_ transaction, right? the name of this and the accessor method should make that clear. Or assign unconditionally.